examples/dice/instrumented/rolldice.go

```diff
@@ -19,42 +22,145 @@ import (
 const name = "go.opentelemetry.io/contrib/examples/dice"
 
 var (
-	tracer  = otel.Tracer(name)
-	meter   = otel.Meter(name)
-	logger  = otelslog.NewLogger(name)
-	rollCnt metric.Int64Counter
+	tracer         = otel.Tracer(name)
+	meter          = otel.Meter(name)
+	logger         = otelslog.NewLogger(name)
+	rollCnt        metric.Int64Counter
+	outcomeHist    metric.Int64Histogram
+	lastRollsGauge metric.Int64ObservableGauge
+	lastRolls      atomic.Int64
 )
 
 func init() {
 	var err error
 	rollCnt, err = meter.Int64Counter("dice.rolls",
-		metric.WithDescription("The number of rolls by roll value"),
+		metric.WithDescription("The number of rolls"),
 		metric.WithUnit("{roll}"))
 	if err != nil {
 		panic(err)
 	}
+
+	outcomeHist, err = meter.Int64Histogram(
+		"dice.outcome",
+		metric.WithDescription("Distribution of dice outcomes (1-6)"),
+		metric.WithUnit("{count}"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	lastRollsGauge, err = meter.Int64ObservableGauge(
+		"dice.last.rolls",
+		metric.WithDescription("The last rolls value observed"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Register the gauge callback.
+	_, err = meter.RegisterCallback(
+		func(ctx context.Context, o metric.Observer) error {
+			o.ObserveInt64(lastRollsGauge, lastRolls.Load())
+			return nil
+		},
+		lastRollsGauge,
+	)
+	if err != nil {
+		panic(err)
+	}
 }
 
-func rolldice(w http.ResponseWriter, r *http.Request) {
-	ctx, span := tracer.Start(r.Context(), "roll")
-	defer span.End()
+func handleRolldice(w http.ResponseWriter, r *http.Request) {
+	// Parse query parameters.
+	rollsParam := r.URL.Query().Get("rolls")
+	player := r.URL.Query().Get("player")
+	maxRolls := 1000 // Arbitrary limit to prevent Slice memory allocation with excessive size value.
+
+	// Default rolls = 1 if not defined.
+	if rollsParam == "" {
+		rollsParam = "1"
+	}
 
-	roll := 1 + rand.Intn(6) //nolint:gosec // G404: Use of weak random number generator (math/rand instead of crypto/rand) is ignored as this is not security-sensitive.
+	// Check if rolls is a number.
+	rolls, err := strconv.Atoi(rollsParam)
+	if err != nil {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusBadRequest)
+		msg := "Parameter rolls must be a positive integer"
+		_ = json.NewEncoder(w).Encode(map[string]string{
+			"status":  "error",
+			"message": msg,
+		})
+		logger.WarnContext(r.Context(), msg)
+		return
+	}
+
+	results, err := rollDice(r.Context(), rolls)
+	if err != nil || rolls > maxRolls {
+		// Signals invalid input (<=0).
+		w.WriteHeader(http.StatusInternalServerError)
+		logger.ErrorContext(r.Context(), err.Error())
```

Comment:
Since the condition `if err != nil || rolls > maxRolls {` implies that `err` might be `nil`.

Of course, once `maxRolls` is removed, this issue is naturally resolved.